src/njoy21/input/MIXR/Card3/src/readMatnWtnList.hpp

```diff
@@ -0,0 +1,31 @@
+template< typename Istream >
+std::vector< std::pair< Argument< Matn >, Argument< Wtn > > >
+                              readMatnWtnList( Istream& is, const size_t nnin ){
+  std::vector< std::pair< Argument< Matn >, Argument< Wtn > > > matnWtnList;
+  std::vector< int > matnList;
+
+  if( nnin > 10 ){
+    Log::error( "The maximum number of materials allowed is 10." );
+    throw std::exception();
+  }
+
+  for( size_t i = 0; i < nnin; i++ ){
+    auto matn_tmp = argument::extract< MIXR::Card3::Matn >( is );
```

Comment:
`mtn_tmp` and `wtn_tmp` should be defined outside the loop and then just reassigned inside the loop. In this way we avoid the cost of repeatedly constructing those values.